tree_in_context/src/type_analysis.egg

```diff
@@ -40,39 +40,89 @@
       ((panic "arg type mismatch"))
       :ruleset type-analysis)
 
-; Pushing expected arg types down
```

Comment:
Why these pushing down rules no longer needed?

Reply:
We are now going bottom-up instead of top-down